runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1307,6 +1307,24 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	 */
 	void fixupInternalLeafPointersAfterCopy(J9IndexableObject *destinationPtr, J9IndexableObject *sourcePtr);
 
+	/**
+	 * Check if the arraylet data is adjacent to the header.
+	 * Mostly used for the detection of camouflaged contiguous arrays that exist with virtualLargeObjectHeap enabled.
+	 *
+	 * @param dataSizeInBytes the size of data in an indexable object, in bytes, including leaves and alignment padding
+	 * @return true if the arraylet data is adjacent to the header, false otherwise
+	 */
+	bool isDataAdjacentToHeader(J9IndexableObject *arrayPtr);
+
+	/**
+	 * Check if the arraylet data is adjacent to the header.
+	 * Mostly used for the detection of camouflaged contiguous arrays that exist with virtualLargeObjectHeap enabled.
```

Comment:
Change to: Used to determine if the array data should be adjacent to its header or in offheap. The determination is based on the size, same how it would be done during the allocation of an object of such a size.